src/eduid_userdb/group_management/state.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+
+import copy
+import datetime
+from dataclasses import asdict, dataclass, field, fields
+from typing import Mapping, MutableMapping, Optional
+
+import bson
+
+from eduid_userdb.exceptions import UserDBValueError
+
+__author__ = 'lundberg'
+
+
+@dataclass(frozen=True)
+class GroupInviteState:
+    group_scim_id: str
+    email_address: str
+    role: str
```

Comment:
could role be an enum?

Reply:
Feels like there are enough enums in both groupdb and in group_management.